src/gwt/tools/generate-prefs.R

```diff
@@ -177,30 +183,91 @@ generate <- function (schemaPath, className) {
          "   /**\n",
          "    * ", def[["description"]], "\n",
          "    */\n")
-      
-      # Add a Java accessor for the preference, and an entry for syncing it with another copy
+
+      # Create a Java/GWT Properties file header for the preference
+      commentProperties <- paste0(
+         "# ", def[["description"]], "\n")
+
+      # Add title and description entries for the java constants file
       prefTitle <- if (is.null(def[["title"]])) "" else def[["title"]]
+      javaConstants <- paste0(javaConstants,
+         comment,
+         "   @DefaultStringValue(\"", prefixDefault, prefTitle, "\")\n",
+         "   String ", camel, "Title", "();\n",
+         "   @DefaultStringValue(\"", prefixDefault, def[["description"]], "\")\n",
+         "   String ", camel, "Description", "();\n")
+
+      if (!is.null(def[["enumReadable"]]))
+      {
+         javaConstants <- paste0(javaConstants,
+                                 paste0(mapply(function (enumVal, enumReadable) {
+                                    paste0(
+                                       "   @DefaultStringValue(\"", prefixDefault, enumReadable, "\")\n",
+                                       "   String ", camel, "Enum_", gsub("[^A-Za-z0-9_]", "_", enumVal), "();"
+                                    )
+                                 }, def[["enum"]], def[["enumReadable"]]), collapse = "\n"
+                                 ), "\n"
+         )
+      }
+
+      javaConstants <- paste0(javaConstants, "\n")
+
+      # Add title and description entries for the java properties file
+      javaProperties <- paste0(javaProperties,
+         commentProperties,
+         camel, "Title = ", prefixProperties, prefTitle, "\n",
+         camel, "Description = ", prefixProperties, def[["description"]], "\n")
+
+      if (!is.null(def[["enumReadable"]]))
+      {
+         javaProperties <- paste0(javaProperties,
+                                  paste0(mapply(function (enumVal, enumReadable) {
+                                     paste0(
+                                        camel, "Enum_", gsub("[^A-Za-z0-9_]", "_", enumVal), "=", prefixProperties, enumReadable
+                                     )
+                                  }, def[["enum"]], def[["enumReadable"]]), collapse = "\n"
+                                  ), "\n"
+         )
+      }
+
+      javaProperties <- paste0(javaProperties, "\n")
+
+      # DEBUG: Add entry here for the java constants and properties files.  Then change perfTitle to
```

Comment:
Does this still need to be done?

Reply:
I believe this is a left over comment from Andrew's work on this. All files are being generated, so I'll remove it